csi/node_server.go

```diff
@@ -31,28 +32,65 @@ func (ns *NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 	}
 
 	targetPath := req.GetTargetPath()
+	devicePath := filepath.Join("/dev/longhorn/", req.GetVolumeId())
```

Comment:
Need to use volume.Endpoint instead. And make sure volume is using `blockdev` frontend.